spec/shopify_transporter/exporters/magento/sql_spec.rb

```diff
@@ -6,23 +6,42 @@ module Exporters
     module Magento
       RSpec.describe SQL do
         context ('#connect') do
-          it 'calls Sequel with the right parameters' do
-            sql_client = SQL.new(
+
+          let(:sql_client) do
+            SQL.new(
               database: 'magento',
               host: 'magento-instance.domain.com',
               port: 1234,
               user: 'dbuser',
               password: 'some_password'
             )
+          end
 
+          let(:connection) { double('connection') }
+
+          it 'calls Sequel with the right parameters' do
             expect(Sequel).to receive(:connect).with(
               adapter: :mysql2,
               database: 'magento',
               host: 'magento-instance.domain.com',
               port: 1234,
               user: 'dbuser',
               password: 'some_password'
-            )
+            ).and_return(connection)
+
+            sql_client.connect { |db| nil }
+          end
+
+          it 'disconnects the database even if the given block raises an error' do
+            expect(Sequel).to receive(:connect).with(
+              adapter: :mysql2,
+              database: 'magento',
+              host: 'magento-instance.domain.com',
+              port: 1234,
+              user: 'dbuser',
+              password: 'some_password'
+            ).and_yield(connection)
+
```

Comment:
This isn't raising an error right now.

Reply:
a remnant of the last test, shouldn't raise an error, changed the title.